hail/src/main/scala/is/hail/expr/ir/IR.scala

```diff
@@ -126,33 +126,20 @@ final case class Coalesce(values: Seq[IR]) extends IR {
   require(values.nonEmpty)
 }
 
+final case class If(cond: IR, cnsq: IR, altr: IR) extends IR
+
 object If {
-  def unify(cond: IR, cnsq: IR, altr: IR, unifyType: Option[Type] = None): If = {
-    if (cnsq.typ == altr.typ)
+  def unify(cond: IR, cnsq: IR, altr: IR, unifyType: Type): If = {
```

Comment:
Needed for use in PruneDeadFields, required for some unifications, like that exemplified in test_mt_full_outer_join